### PR TITLE
 docs: [FC-0074] document contribution process to the hooks extension framework 

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description

This PR adds an overview of the frameworks contribution process from conception to contribution. The conception section is repo-specific, so it'd later on written as part of the "How To Create A(n) Event/Filter" for each repository, as for now, we're using the available documentation.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.


#### :radio_button: Update the status of your PR

Your PR is currently marked as a draft.  After completing the steps above, update its status by clicking "Ready for Review", or removing "WIP" from the title, as appropriate.




#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/docs-openedx-org-maintainers`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

File: source/developers/concepts/hooks_extension_framework.rst
Comment (mariajgrimaldi) [inline source/developers/concepts/hooks_extension_framework.rst:166]: I'm still thinking this is missing something. I'll re-review it tomorrow so I can open it for others to review.

Comment (mariajgrimaldi) [reply source/developers/concepts/hooks_extension_framework.rst:166]: I think the high-level overview of the contribution process to the framework is good as it is.

File: source/developers/concepts/hooks_extension_framework.rst
Comment (sarina) [inline source/developers/concepts/hooks_extension_framework.rst:169]: I think I would not nest this in a "What's Next" - I would just have this be its own second-level header so it shows up in the sidebar (so with `****` below)

File: source/developers/concepts/hooks_extension_framework.rst
Comment (sarina) [inline source/developers/concepts/hooks_extension_framework.rst:172]: I don't think I'd recommend making a proposal in Slack - proposals in the forum or issue tracker are better.

File: source/developers/concepts/hooks_extension_framework.rst
Comment (sarina) [inline source/developers/concepts/hooks_extension_framework.rst:188]:
```suggestion
#. Once you've validated both implementations, you can open the Pull Requests for review. Typically, the Pull Request for the hook definition (in `openedx-events`_ or `openedx-filters`_) is opened first, along with a simultaneous PR in the service. While managing two Pull Requests at the same time might be challenging, it's recommended to avoid rework by validating definitions early, like ensuring the filter arguments are objects available during execution, verifying the event payload can be populated with the available information, that the hook name fits the context where it will be used, etc.
```

Readability suggestion

**Review by sarina — APPROVED** [review]
Some small suggestions, looks great!

File: source/developers/concepts/hooks_extension_framework.rst
Comment (BryanttV) [inline source/developers/concepts/hooks_extension_framework.rst:174]: In some places, the default link text I think is not the best, for example:
image

This would look better:
image

**Review by BryanttV — COMMENTED** [review]
Hi @mariajgrimaldi, this looks good! The instructions for the contribution process are quite clear and detailed. I only have a little comment.

**Comment by mariajgrimaldi** [comment]
@sarina @BryanttV: comments addressed, thank you!

**Review by BryanttV — APPROVED** [review]

**Comment by sarina** [comment]
@mariajgrimaldi is this ready to merge?

**Comment by mariajgrimaldi** [comment]
@sarina: yes, I'd say so.

**Pull request merged**
